app/controllers/activitypub/collections_controller.rb

```diff
@@ -10,7 +10,7 @@ class ActivityPub::CollectionsController < ActivityPub::BaseController
   before_action :set_cache_headers
 
   def show
-    expires_in 3.minutes, public: public_fetch_mode?
+    expires_in 3.minutes, public: public_fetch_mode? && signed_request_account.nil?
```

Comment:
This effectively disables caching of the pinned toots, which increases load when an account is newly-discovered (or refreshed) by multiple instances (e.g., when one of your toots get boosted by someone with lots of users on instances which don't follow you).

Since pinned toots are public, I'd suggest that instead of not caching when there is an user, not caching when the user is *blocked*.